### PR TITLE
Print connection ID

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
To help debug abnormal phenomena found in #31943 that EDS comes before CDS

### CODE REVIEW

**Review by adiprerepa — APPROVED** [review]
Lgtm

File: pilot/pkg/xds/gen.go [hunk 1]
```diff
@@ -128,10 +128,10 @@ func (s *DiscoveryServer) pushXds(con *Connection, push *model.PushContext,
 		if log.DebugEnabled() {
 			// Add additional information to logs when debug mode enabled
 			log.Infof("%s: PUSH%s for node:%s resources:%d size:%s nonce:%v version:%v",
-				v3.GetShortType(w.TypeUrl), req.PushReason(), con.proxy.ID, len(res), util.ByteCount(ResourceSize(res)), resp.Nonce, resp.VersionInfo)
+				v3.GetShortType(w.TypeUrl), req.PushReason(), con.ConID, len(res), util.ByteCount(ResourceSize(res)), resp.Nonce, resp.VersionInfo)
```
Comment (howardjohn) [inline pilot/pkg/xds/gen.go:131]: I am not sure this is not too noisy, is the connection id pretty long?

Comment (hzxuzhonghu) [reply pilot/pkg/xds/gen.go:131]: nope, just proxyId + connection number

**Comment by howardjohn** [comment]
I am not sure the eds could be from previous connection anyways, we logged that it disconnected already?

File: pilot/pkg/xds/gen.go (hunk 1 above)
Comment (howardjohn) [reply pilot/pkg/xds/gen.go:131]: it looks like it repeats proxyid multiple times. I don't think a log would fit on a single line on a large monitor anymore which makes the logs hard to see

File: pilot/pkg/xds/ads.go [hunk 2]
```diff
@@ -488,15 +488,14 @@ func listEqualUnordered(a []string, b []string) bool {
...
-	con.ConID = connectionID(node.Id)
-	con.node = node
-
 	// Setup the initial proxy metadata
 	proxy, err := s.initProxyMetadata(node)
 	if err != nil {
 		return err
 	}
+	// First request so initialize connection id and start tracking it.
+	con.ConID = connectionID(proxy.ID)
```
Comment (hzxuzhonghu) [inline pilot/pkg/xds/ads.go:497]: @howardjohn I missed the proxy id and node id previously, this is what we want.

File: pilot/pkg/xds/gen.go (hunk 1 above)
Comment (hzxuzhonghu) [reply pilot/pkg/xds/gen.go:131]: acked

File: pilot/pkg/xds/gen.go [hunk 3]
```diff
@@ -128,10 +128,10 @@ func (s *DiscoveryServer) pushXds(con *Connection, push *model.PushContext,
...
 		} else {
 			log.Infof("%s: PUSH%s for node:%s resources:%d size:%s",
-				v3.GetShortType(w.TypeUrl), req.PushReason(), con.proxy.ID, len(res), util.ByteCount(ResourceSize(res)))
+				v3.GetShortType(w.TypeUrl), req.PushReason(), con.ConID, len(res), util.ByteCount(ResourceSize(res)))
```
Comment (howardjohn) [inline pilot/pkg/xds/gen.go:134]: sds.go and eds.go needs the same change

**Review by howardjohn — COMMENTED** [review]
LGTM except eds

File: pilot/pkg/xds/gen.go (hunk 3 above)
Comment (hzxuzhonghu) [reply pilot/pkg/xds/gen.go]: Th Generate is a general method, i could not change it. Just changed the log format.

**Pull request merged**